runtime/memory.c

```diff
@@ -525,40 +513,46 @@ Caml_inline value caml_alloc_shr_aux (mlsize_t wosize, tag_t tag, int track,
 }
 
 #ifdef WITH_PROFINFO
+  /* Use this to debug problems with macros... */
+  #define NO_PROFINFO 0xff
+  #define Copy_profinfo(pf) pf
+#else
+  #define NO_PROFINFO 0
+  #define Copy_profinfo(pf) NO_PROFINFO
+#endif
```

Comment:
Could we just ensure in `caml/mlvalues.h` that `Profinfo_hd` is also defined when `WITH_PROFINFO` is not set, and returns an appropriate `NO_PROFINFO` value in this case? Or do we *not* want that because potential programming errors (calling `Profinfo_hd` outside a WITH_PROFINFO check) would go unnoticed?

Reply:
I did it like this to avoid changing the public API by introducing the Profinfo_hd macro when WITH_PROFINFO is not set. I don't have a strong opinion either way, though.

Reply:
I've just noticed that `caml_alloc_shr_with_profinfo` is always defined regardless of whether `WITH_PROFINFO` is true, so it's not like there's a serious effort to ensure profinfo symbols are in scope only if configured. So I think it's fine to add `Profinfo_hd` unconditionally, and I agree that would simplify things.